AIPscan/helpers.py

```diff
@@ -10,6 +11,21 @@ def parse_bool(val, default=True):
         return default
 
 
+def parse_date(date_string):
+    """Parse date parameter string into datetime object or None.
+
+    :param date_string: Date string. Valid format is YYYY-MM-DD.
+
+    :returns: datetime.datetime object or None
```

Comment:
I might consider doing the work of `_prepare_datetimes_for_query(...)` here and returning a consistent data type in this helper. For me there are two main clues/reasons: 

1. Mixing return types might cause complexity upstream. Testing for `None` vs. getting to work on a newly minted `datetime`. 
2. I appreciate the amount of thought gone into the date handling across the PR, but the story as I see it through the code, is that we're largely only dealing with date ranges which I believe these helpers are largely setup for handling going forward. I think we're always going to know with a min or a max. So I might suggest a function signature of, something like:

```python
def parse_datetime_bounds(date_string, Upper=False):
   return datetime (or datetime + 1 day if upper==true) if valid,
   else return datetime.min if invalid, or datetime.max if invalid and upper==true
```
I think we might see a cleaner coding style appear by decoding early, rather than doing a little bit now, and a little bit later.